### PR TITLE
2019.10.2/add strict flag

### PR DESCRIPTION
**Pull request by ian-heinzman**
Adding a -s flag that validates the target ("Task") programs are present in the filesystem. This way you can catch potentially bad configs during composition rather than runtime.

### CODE REVIEW

File: alyeska/compose/compose_sh.py [hunk 1]
```diff
@@ -166,7 +167,21 @@ def convert_yaml_to_sh(
...
+        prefab_flags (List, optional): A list of flags to parse. Useful for testing.
+
+    Returns:
+        None -- the now-parsed flags can be accessed through the FLAGS variable.
+    """
+    global FLAGS
+
+    # Don't re-parse flags
+    if FLAGS:
+        return
```
Comment (vogt4nick) [inline alyeska/compose/compose_sh.py]: More explicit:
```suggestion
        raise NotImplementedError("Cannot parse flags more than once.")
```

What do you think about raising an error if flags have already been parsed? This exit may cause some non-obvious behavior as is; e.g. If init_flags is called a second time with different arguments, the user won't know the flags haven't been updated.

File: alyeska/compose/compose_sh.py [hunk 2]
```diff
@@ -166,7 +167,21 @@ def convert_yaml_to_sh(
     return shell_script
 
 
-def main():
+def init_flags(prefab_flags: List = None) -> None:
```
Comment (vogt4nick) [inline alyeska/compose/compose_sh.py]: I'm glad you moved the arg parsing to its own function. 👍

File: tests/compose/test_compose_sh.py [hunk 3]
```diff
@@ -16,16 +16,12 @@
...
-import pytest
-
-from alyeska.compose.compose_sh import convert_yaml_to_sh
+from alyeska.compose.compose_sh import init_flags, convert_yaml_to_sh
 
 from test_compose_globals import COMPOSE_SMALL, COMPOSE_BIG, COMPOSE_CYCLE
 
 
 def test__convert_yaml_to_sh():
+    init_flags(["foo"])  # Necessary to initialize the flags object.
```
Comment (vogt4nick) [inline tests/compose/test_compose_sh.py]: Do we want to raise an error if `init_flags` receives an empty list?

**Comment by vogt4nick** [comment]
I realize we also need to version bump this guy.

File: alyeska/compose/compose_sh.py (hunk 1 above)
Comment (ian-heinzman) [reply alyeska/compose/compose_sh.py]: Fair point. Done.

File: tests/compose/test_compose_sh.py (hunk 3 above)
Comment (ian-heinzman) [reply tests/compose/test_compose_sh.py]: It already does since config_file is a required argument.

**Comment by ian-heinzman** [comment]
Version is now bumped.

File: alyeska/compose/compose_sh.py [hunk 4]
```diff
@@ -179,10 +192,22 @@ def main():
...
         "-o", action="store", dest="ofile", default="compose.sh", type=pathlib.Path
     )
-    parser.add_argument("-v", action="store_true", dest="verbose_output", default=False)
-    args = parser.parse_args()
+    parser.add_argument(
+        "-s",
+        action="store_true",
+        dest="strict",
+        help="Causes the utility to fail if any task file is not found",
+    )
```
[the remark this answers is not in the document]
Comment (ian-heinzman) [reply alyeska/compose/compose_sh.py]: SGTM. Named the flag `--no-check` though.

**Pull request merged**